framework/src/vectorpostprocessors/SamplerBase.C

```diff
@@ -44,104 +44,106 @@ void
 SamplerBase::setupVariables(const std::vector<std::string> & variable_names)
 {
   _variable_names = variable_names;
+  _values.reserve(variable_names.size());
 
-  _values.resize(variable_names.size());
-  _values_tmp.resize(variable_names.size());
-
-  for (unsigned int i=0; i<variable_names.size(); i++)
-    _values[i] = &_vpp->declareVector(variable_names[i]);
+  for (const auto & variable_name : variable_names)
+    _values.push_back(&_vpp->declareVector(variable_name));
```

Comment:
`emplace_back`?  I know it doesn't matter much here... but I've started using it for essentially everything...